dandiapi/api/services/dandiset/exceptions.py

```diff
@@ -5,3 +5,10 @@
 
 class DandisetAlreadyExistsError(DandiError):
     http_status_code = status.HTTP_400_BAD_REQUEST
+
+
+class UnauthorizedEmbargoAccess(DandiError):
+    http_status_code = status.HTTP_401_UNAUTHORIZED
+    message = (
+        "Authentication credentials must be provided when attempting to access embargoed dandisets"
```

Comment:
while troubleshooting this I realized we nowhere show how to authenticate for such requests.  I wonder if it would make sense to add here or not? e.g.

```suggestion
        "Authentication credentials must be provided in the 'Authorization' header field with 'token API_TOKEN' value when attempting to access embargoed dandisets"
```

or alike. WDYT?

Reply:
The response of any one API endpoint isn't really the place for documentation about authentication to our API. That would go in some kind of specific documentation, like the [handbook](https://www.dandiarchive.org/handbook/40_development/).

If what you're saying is that we need better documentation around API authentication, that's a fair point, but is separate from the concerns of this PR.

Reply:
agreed!

Reply:
filed https://github.com/dandi/handbook/issues/109